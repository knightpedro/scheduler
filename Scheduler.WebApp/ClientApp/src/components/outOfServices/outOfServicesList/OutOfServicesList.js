import React, { useMemo } from "react";
import { Link, generatePath } from "react-router-dom";
import Routes from "../../../routes";
import moment from "moment";
import { BasicTable, SelectColumnFilter } from "../../common/tables";

const TIME_FORMAT = "HH:mm DD/MM/YYYY";

const OutOfServicesList = ({ outOfServices }) => {
  const createPath = id => generatePath(Routes.outOfServices.EDIT, { id });

  const columns = useMemo(
    () => [
      {
        Header: "Reason",
        accessor: "reason",
        Cell: ({ row }) => (
          <Link to={createPath(row.original.id)}>{row.values.reason}</Link>
        ),
        Filter: SelectColumnFilter
      },
      {
        Header: "Start",
        accessor: "start",
        className: "text-center",
        sortType: "momentSort",
        disableFilters: true,
        Cell: ({ cell: { value } }) => value.format(TIME_FORMAT)
      },
      {
        Header: "End",
        accessor: "end",
        className: "text-center",
        sortType: "momentSort",
        disableFilters: true,
        Cell: ({ cell: { value } }) => value.format(TIME_FORMAT)
      }
    ],
    []
  );

  const data = useMemo(
    () =>
      outOfServices.map(outOfService => ({
        id: outOfService.id,
        reason: outOfService.description,
        start: moment(outOfService.start),
        end: moment(outOfService.end)
      })),
    [outOfServices]
  );

  const initialTableState = useMemo(
    () => ({
      sortBy: [
        {
          id: "start",
          desc: true
        }
      ]
    }),
    []
  );

  return (
    <BasicTable
      columns={columns}
      data={data}
      initialState={initialTableState}
    />
  );
};

export default OutOfServicesList;
