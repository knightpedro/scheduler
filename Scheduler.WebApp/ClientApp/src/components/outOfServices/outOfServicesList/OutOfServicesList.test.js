import React from "react";
import { shallow } from "enzyme";
import OutOfServicesList from "./OutOfServicesList";

const oosStub = [
    {
        id: 1,
        description: "Damage",
        start: "2020-02-29T19:00:00.0000000Z",
        end: "2020-02-29T23:00:00.0000000Z",
    },
];

describe("OutOfServicesList", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<OutOfServicesList outOfServices={oosStub} />);
        expect(wrapper).toMatchSnapshot();
    });
});
