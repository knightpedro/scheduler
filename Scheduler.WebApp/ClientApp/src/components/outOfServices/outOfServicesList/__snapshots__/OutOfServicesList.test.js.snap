// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutOfServicesList renders correctly 1`] = `
<BasicTable
  columns={
    Array [
      Object {
        "Cell": [Function],
        "Filter": [Function],
        "Header": "Reason",
        "accessor": "reason",
      },
      Object {
        "Cell": [Function],
        "Header": "Start",
        "accessor": "start",
        "className": "text-center",
        "disableFilters": true,
        "sortType": "momentSort",
      },
      Object {
        "Cell": [Function],
        "Header": "End",
        "accessor": "end",
        "className": "text-center",
        "disableFilters": true,
        "sortType": "momentSort",
      },
    ]
  }
  data={
    Array [
      Object {
        "end": "2020-02-29T23:00:00.000Z",
        "id": 1,
        "reason": "Damage",
        "start": "2020-02-29T19:00:00.000Z",
      },
    ]
  }
  initialState={
    Object {
      "sortBy": Array [
        Object {
          "desc": true,
          "id": "start",
        },
      ],
    }
  }
/>
`;
